### PR TITLE
Feature/update read write ck file

### LINKED ISSUE #316: Update and test CK read/write routines
**Describe your use case**
The SPICE Camera Kernel (CK) read and write routines are outdated and untested. This issue addresses these.

**Describe alternatives solutions you've considered**
We may switch to the ``spiceypy`` library instead of our custom C-interfacing routines.

**Additional context**
N/A.

### PR DESCRIPTION
**Pull request by joaogvcarneiro**
* **Tickets addressed:** #316 
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

The first four commits refactor the SPICE CK routines, the fifth adds the MAVEN spacecraft spacecraft clock and the sixth adds the test file.

## Verification

The test file runs a simulation, writes the rotational data to a CK file, reads it and compares the output to the original simulation data.

## Documentation

No documentation is added, but a significant amount of comments are included for future developers.

## Future work

Might change the routines to use ``spiceypy`` instead.

### CODE REVIEW

File: src/utilities/pyswice_ck_utilities.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 from Basilisk.utilities import RigidBodyKinematics, macros
 
 
-def ckWrite(handle, time, MRPArray, avArray, startSeg, sc = -62, rf = "J2000"):
+def ckWrite(handle, time, MRPArray, avArray, startSeg, sc=-62, rf="J2000"):
```
Comment (thibaudteil) [inline src/utilities/pyswice_ck_utilities.py]: while we are here, let's make these variable names more verbose. Something like "spacecraft_id" or "reference_frame" .
We should also snake case everywhere

File: src/utilities/pyswice_ck_utilities.py [hunk 2]
```diff
@@ -103,42 +104,42 @@ def ckRead(time, SCID=-62, rf="J2000"):
...
-        kernalQuatArray[0, s] = -kernalQuat[s]
-        if s == 0:
-            kernalQuatArray[0, s] = -kernalQuatArray[0, s]
-    etout = pyswice.doubleArray_getitem(et, 0)
-    return etout, kernalQuatArray, kernOmega
+    for i in range(9):
+        cmatConversion[i // 3, i % 3] = pyswice.doubleArray_getitem(cmat, i)  # Map 9D array into 3x3 matrix
+
+    # Convert attitude to quaternions
+    kernQuat = RigidBodyKinematics.C2EP(cmatConversion)
```
Comment (thibaudteil) [inline src/utilities/pyswice_ck_utilities.py]: I think we should start importing the cpp rigidBodyKinematics library (in architecture/utilities) which is identical in functionality but has different function names

**Review by thibaudteil — APPROVED** [review]
Just some small formatting changes

File: src/utilities/pyswice_ck_utilities.py (hunk 1 above)
Comment (joaogvcarneiro) [reply src/utilities/pyswice_ck_utilities.py]: Done!

File: src/utilities/pyswice_ck_utilities.py (hunk 2 above)
Comment (joaogvcarneiro) [reply src/utilities/pyswice_ck_utilities.py]: Hmm, but we're doing the conversions in Python, right? We can't use the C++ Rigid Body Kinematics library, can we?

Comment (thibaudteil) [reply src/utilities/pyswice_ck_utilities.py]: It's wrapped, so you should be able to import it 

```
from Basilisk.architecture import rigidBodyKinematics

kernQuat = rigidBodyKinematics.dcmToEulerParameter(cmatConversion)
```
Type of thing

**Pull request merged**